lib/src/git_subprocess.rs

```diff
@@ -0,0 +1,484 @@
+use std::ffi::OsStr;
+use std::fmt;
+use std::num::NonZeroU32;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process::Command;
+use std::process::Output;
+use std::process::Stdio;
+
+use thiserror::Error;
+
+use crate::git::GitFetchError;
+use crate::git::GitPushError;
+
+/// New-type for comparable io errors
+pub struct IoError(std::io::Error);
+impl PartialEq for IoError {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.kind().eq(&other.0.kind())
+    }
+}
```

Comment:
Perhaps, we should remove `PartialEq` from `GitPushError` instead. I'll take a look.